### ISSUE TITLE
`derive(Component)` appears to be incompatible with `missing_docs` lint

### ISSUE THREAD

**Issue by EricHripko** [Mcat12/shaku#26]
# Overview
I'm trying to ensure that all public API of my crate is documented, so I enabled `missing_docs` lint. Unfortunately, it seems that `shaku` doesn't attach docs to any generated a `...Parameters` structs. I've also not found any way to attach docs to such structs manually.

# Expected Behaviour
Ideally, I'd hope that `shaku`-generated code can pass `missing_docs` check.

# Actual Behaviour
Each component that requires parameters results in a warning from `missing_docs` check.
```
warning: missing documentation for a struct
  --> src\lib.rs:35:10
   |
35 | #[derive(Component)]
   |          ^^^^^^^^^
   |
   = note: this warning originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by Mcat12** [comment]
Are you trying to document the parameters struct or exclude it from the lint check?

If you want to exclude it from the lint check, a temporary solution would be to wrap the components in a private module that allows missing docs:
```rust
#![deny(missing_docs)]

pub use components::*;
mod components {
    #![allow(missing_docs)]
    
    /// Component docs here
    #[derive(Comenent)]
    pub struct // ...
}
```

In the current version of shaku I don't think you can add documentation to a generated parameters struct. If you want to document the parameters struct, you should probably write the `Component` implementation yourself:
```rust
/// Component docs here
pub struct MyComponent {
    // ...
}

/// Parameters docs here
pub struct MyComponentParameters {
    // ...
}

impl<M: Module> Component for MyComponent {
    type Parameters = MyComponentParameters;
    // ...
}
```

For future versions of shaku, some way to add documentation to the parameters struct when using the derive macro could be added. The question is how the API for that should look. How would you expect it to look? I have some ideas, but they all have various issues (like using the `///` syntax might not be viable or ergonomic). Maybe the parameters struct could be annotated with `/// Parameters for {component_name}` and each field would have the docs from the component's field copied over? (The issue with this is that it's not flexible.)

**Comment by EricHripko** [comment]
Thank you for the workaround - this allowed me to enable `missing_docs` check with the current version of `shaku` 🙌 I think auto-generating docs (in somewhat opinionated way) would be a solid starting point and (personally) I'd be happy with the approach you outlined 🙂

**Comment by Mcat12** [comment]
The opinionated solution I outlined is now implemented in shaku_derive 0.6.1

**Comment by EricHripko** [comment]
That's awesome - thank you for taking a look at this 🙌 Bumping to `0.6.1` enabled me to remove the `#![allow(missing_docs)]` workaround 🙂
